torch/csrc/api/include/torch/nn/functional/activation.h

```diff
@@ -20,6 +20,15 @@ inline Tensor hardshrink(const Tensor& input,
   return torch::hardshrink(input, options.lambda());
 }
 
+inline Tensor hardtanh(const Tensor& input, const HardtanhOptions& options) {
+  if (options.inplace()) {
+    return torch::hardtanh_(const_cast<Tensor&>(input),
```

Comment:
same here: we might want to just change `input`'s type to `Tensor&`, and remove `const_cast`.